action.yml

```
@@ -20,6 +24,16 @@ runs:
       with:
         java-version: ${{ inputs.java-version }}
         distribution: ${{ inputs.distribution }}
+    - name: Set up Gradle Dependencies Cache
+      if: ${{ inputs.cache != 'false' }}
+      uses: actions/cache@v3
+      with:
+      path: |
+        ~/.gradle/caches
+        ~/.gradle/wrapper
+      key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+      restore-keys: |
+        ${{ runner.os }}-gradle-
```

```suggestion
        cache: ${{ inputs.cache }}
```


```yml
inputs:
  cache:
    description: 'Name of the build platform to cache dependencies. It can be "maven", "gradle" or "sbt".'
    required: false
    default: 'gradle'
```

Or you can use [actions/setup-java](https://github.com/actions/setup-java#caching-packages-dependencies). `setup-java@v3` or newer has caching **built-in**.